models/project_issue.go

```diff
@@ -181,33 +182,36 @@ func addUpdateIssueProject(e *xorm.Session, issue *Issue, doer *User, newProject
 // |_|   |_|  \___// |\___|\___|\__|____/ \___/ \__,_|_|  \__,_|
 //               |__/
 
-// MoveIssueAcrossProjectBoards move a card from one board to another
-func MoveIssueAcrossProjectBoards(issue *Issue, board *ProjectBoard) error {
+// MoveIssuesOnProjectBoard moves or keeps issuses in a column and sorts them inside of that column
+func MoveIssuesOnProjectBoard(board *ProjectBoard, issues map[int64]*Issue) error {
 	sess := db.NewSession(db.DefaultContext)
 	defer sess.Close()
 	if err := sess.Begin(); err != nil {
 		return err
 	}
 
-	var pis ProjectIssue
-	has, err := sess.Where("issue_id=?", issue.ID).Get(&pis)
-	if err != nil {
-		return err
-	}
+	for sorting, issue := range issues {
+		var pis ProjectIssue
+		has, err := sess.Where("issue_id=?", issue.ID).Get(&pis)
```

Comment:
Maybe we can `count distinct where project_id=? and issue_id IN (...)` to see whether the count is the same as the issue list.  If the count is the same, then all issues are in this project, it will be faster.